### PR TITLE
``pyreverse``: better error messages for unsupported file formats

### LINKED ISSUE #5950: DOC: pyreverse supported format not recognized missing puml, plantuml, mmd, etc
### Bug description

Some supported formats are not listed in the error messaging:

```
> pyreverse -ASmy -o .puml my_package/my_module.py 
parsing my_package/my_module.py...
Format: "puml" not recognized. Use one of: bmp canon cmap cmapx cmapx_np dot dot_json eps fig gd gd2 gif gtk gv ico imap imap_np ismap jpe jpeg jpg json json0 mp pdf pic plain plain-ext png pov ps ps2 svg svgz tif tiff tk vml vmlz vrml wbmp x11 xdot xdot1.2 xdot1.4 xdot_json xlib
```

Here, I have made the mistake of prepending a `.` to `puml`. The error output should list `puml` (and `plantuml` and `mmd` and `vcg`) as [supported formats](https://pylint.pycqa.org/en/v2.12.2/additional_commands/index.html?highlight=.puml#example-output).

Without the preceding `.`, the command runs as expected.

### Configuration

_No response_

### Command used

```shell
`> pyreverse -ASmy -o .puml my_package/my_module.py `
```

### Pylint output

```shell
parsing my_package/my_module.py...
Format: "puml" not recognized. Use one of: bmp canon cmap cmapx cmapx_np dot dot_json eps fig gd gd2 gif gtk gv ico imap imap_np ismap jpe jpeg jpg json json0 mp pdf pic plain plain-ext png pov ps ps2 svg svgz tif tiff tk vml vmlz vrml wbmp x11 xdot xdot1.2 xdot1.4 xdot_json xlib
```
```


### Expected behavior

All supported formats are listed when user makes a typo in the `-o` format argument

### Pylint version

```shell
> pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.8.12 | packaged by conda-forge | (default, Oct 12 2021, 21:59:51) 
[GCC 9.4.0]
```
```


### OS / Environment

OpenSUSE Tumbleweed

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by DudeNr33**
- [x] Add yourself to ``CONTRIBUTORS.txt`` if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description
When supplying any format not directly known to ``pyreverse``, we will try to convert it using Graphviz (if available).
Until now we just called the ``dot`` command and presented any error messages from it directly to the user, which can be confusing if it complains about unsupported file formats.

This PR aims to make the error messages clearer and to fail fast if we can already tell that the format is not supported by Graphviz.

Closes #5950

### CODE REVIEW

File: pylint/pyreverse/dot_printer.py [hunk 1]
```diff
@@ -152,7 +152,6 @@ def generate(self, outputfile: str) -> None:
         with open(dot_sourcepath, "w", encoding="utf8") as outfile:
             outfile.writelines(self.lines)
         if target not in graphviz_extensions:
-            check_graphviz_availability()
```
Comment (DudeNr33) [inline pylint/pyreverse/dot_printer.py:155]: We already check that in the ``Run`` class. No need to do it twice.

File: pylint/pyreverse/main.py [hunk 2]
```diff
@@ -139,7 +152,10 @@
             action="store",
             default="dot",
             metavar="<format>",
-            help="create a *.<format> output file if format available.",
+            help=(
+                f"create a *.<format> output file if format is available. Available formats are: {', '.join(DIRECTLY_SUPPORTED_FORMATS)}. "
+                f"Any other format will be tried to create by means of the 'dot' command line tool, which requires a graphviz installation."
+            ),
```
Comment (DudeNr33) [inline pylint/pyreverse/main.py:158]: We did not state the directly supported output formats in the help output previously, I think this also makes it a bit clearer what is going on under the hood.

File: pylint/pyreverse/utils.py [hunk 3]
```diff
@@ -292,7 +293,34 @@ def check_graphviz_availability():
     if shutil.which("dot") is None:
         print(
             "The requested output format is currently not available.\n"
-            "Please install 'Graphviz' to have other output formats "
-            "than 'dot' or 'vcg'."
+            "Please install 'Graphviz' to have image output formats."
```
Comment (DudeNr33) [inline pylint/pyreverse/utils.py:296]: The message was outdated as we now support other output formats as well. Reiterating the directly supported output formats here is not necessary I think, so I kept it more general.

File: pylint/pyreverse/utils.py [hunk 4]
```diff
@@ -292,7 +293,34 @@ def check_graphviz_availability():
...
+        pattern=r".*Use one of: (?P<formats>(\S*\s?)+)",
+        string=dot_output.stderr.strip(),
+    )
+    if not match:
+        print(
+            "Unable to determine Graphviz supported output formats. "
+            "Pyreverse will continue, but subsequent error messages "
+            "regarding the output format may come from Graphviz directly."
+        )
+        return
```
Comment (DudeNr33) [inline pylint/pyreverse/utils.py:320]: ``mypy`` forced me to include this. 😉 
Relying on the textual output of the ``dot`` command to determine the supported output formats is probably not the best solution, but I'd argue the tradeoff for more user friendly error messages and ability to fail fast is worth it. 
But if you have concerns or feedback please feel free to address it.

File: pylint/pyreverse/utils.py (hunk 3 above)
Comment (Pierre-Sassoulas) [reply pylint/pyreverse/utils.py:296]:
```suggestion
            "'Graphviz' need to be installed for your chosen output format."
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Look pretty good already. I think we can include in 2.13 :)

File: tests/pyreverse/test_main.py [hunk 5]
```diff
@@ -29,3 +53,62 @@ def test_project_root_in_sys_path():
...
+@mock.patch("pylint.pyreverse.main.DiadefsHandler", new=mock.MagicMock())
+@mock.patch("pylint.pyreverse.main.writer")
+@pytest.mark.usefixtures("mock_graphviz")
+def test_graphviz_supported_image_format(mock_writer, capsys):
+    """Test that Graphviz is used if the image format is supported."""
+    with pytest.raises(SystemExit):
+        # we have to catch the SystemExit so the test execution does not stop
+        main.Run(["-o", "png", TEST_DATA_DIR])
+    # Check that the right info message is shown to the user
+    assert (
```
[the remark this answers is not in the document]
Comment (DudeNr33) [reply tests/pyreverse/test_main.py]: I included the check.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**